ode_solver.cpp

```diff
@@ -78,7 +80,7 @@ std::pair<Eigen::VectorXd, unsigned int> ODE::newton_method(const FunctionBase &
      * --> x1 = x0 + d
      */
     auto f = fcn.value(x0);
-    Eigen::VectorXd d = jac.solve(-f);
+    d = jac.solve(-f);
```

Comment:
The point here is that in every iteration, you create the variable (=requires memory allocation) and then at the end of the block (the `}`), you destroy the variable again (=deallocates memory).

You can avoid the repeated allocation/deallocation if you move the declaration of the variable outside the loop.